Assignment3/MyDictionary.java

```diff
@@ -0,0 +1,32 @@
+package assignment3;
+
+public class MyDictionary{
+  private Set<String> words;
+  private Set<String> prefixes;
+  
+  public MyDictionary(String[] words){
+   
+    this.words = new HashSet<>(Arrays.asList(words));
+    this.prefixes = new HashSet<>();
+        for (String word : words) {
+            for (int i = 1; i <= word.length(); i++) {
+                this.prefixes.add(word.substring(0, i));
+            }
+        }
+    }
+  
+  public void printPrefixes(){
+    Iterator it = prefixes.iterator();
```

Comment:
Do you know about the normal for loop notation in java?

for(String prefix : prefixes) {
  System.out.println(prefix);
}